includes/LogisticsOrderFactory.php

```diff
@@ -9,21 +9,151 @@
 
 defined('ABSPATH') || exit;
 
+use Towa\GebruederWeissSDK\Model\Address;
+use Towa\GebruederWeissSDK\Model\AddressReference;
+use Towa\GebruederWeissSDK\Model\Article;
+use Towa\GebruederWeissSDK\Model\ArticleNote;
+use Towa\GebruederWeissSDK\Model\Contact;
+use Towa\GebruederWeissSDK\Model\LingualText;
+use Towa\GebruederWeissSDK\Model\LogisticsAddress;
 use Towa\GebruederWeissSDK\Model\LogisticsOrder;
+use Towa\GebruederWeissSDK\Model\LogisticsRequirements;
+use Towa\GebruederWeissSDK\Model\OrderLine;
 
 /**
  * Factory for creating Logistics Orders
  */
 class LogisticsOrderFactory
 {
+    /**
+     * Settings Repository
+     *
+     * @var SettingsRepository
+     */
+    private $settingsRepository;
+
+    /**
+     * Creates an instance of the logistics order factory
+     *
+     * @param SettingsRepository $settingsRepository An instance of the settings repository.
+     */
+    public function __construct(SettingsRepository $settingsRepository)
+    {
+        $this->settingsRepository = $settingsRepository;
+    }
+
     /**
      * Creates a logistics order from a WooCommerce order
      *
      * @param object $wooCommerceOrder The order to be converted into a logistics order.
      * @return LogisticsOrder
      */
-    public static function buildFromWooCommerceOrder(object $wooCommerceOrder): LogisticsOrder
+    public function buildFromWooCommerceOrder(object $wooCommerceOrder): LogisticsOrder
+    {
+        $logisticsOrder = new LogisticsOrder();
+        $logisticsOrder->setUrl($this->settingsRepository->getSiteUrl() . "/wp-json/gebrueder-weiss-woocommerce/v1/update/" . $wooCommerceOrder->get_id());
```

Comment:
we should shift the hardcoded part here into the .env since the callback-endpoint in OrderController is also hardcoded like this

Reply:
Moving it to the .env file is the wrong move in my opinion, since the typical user of the plugin will not configure env variables. If we want to avoid this string, we could define it as a constant in the controller. However, I am fine with a string beeing used here.